blockchain/src/blockchain/rebranch_utils.rs

```diff
@@ -248,7 +262,7 @@ impl Blockchain {
 
             // Include the body of the new block (at position 0).
             self.chain_store
-                .put_chain_info(write_txn, &fork_block.0, &fork_block.1, i == 0);
+                .put_chain_info(write_txn, &fork_block.0, &fork_block.1, i == 0, true);
```

Comment:
This line looks a bit fishy. We're iterating over `target_chain` here, which are the blocks on the fork we're rebranching to, so these blocks are `stored` in the chain store with full body. When `i == 0` and thus `include_body = true`, the respective block will now also be stored as `pushed`, while all other blocks remain `stored`, even though they are now `pushed`. This doesn't really cause any problems as the blocks can still be discovered, but it is inconsistent and leads to data duplication. However, it might actually be faster to duplicate the transactions instead of deleting them.